pipeline_steps/common.groovy

```diff
@@ -2131,14 +2135,84 @@ Boolean willOverlapMaintenanceWindow(){
   LocalDateTime maintWindowStart = getNextMaintenanceWindowStart()
   LocalDateTime prBufferStart = maintWindowStart.minus(prBuffer)
 
-  return prBufferStart.isBefore(completion)
+  Boolean willOverlap = prBufferStart.isBefore(completion)
+  if(willOverlap){
+    print("This build is predicted to overlap the next maintenance window")
+  }
+  return willOverlap
 }
 
+
+
 Boolean issueExistsForNextMaintenanceWindow(project="RE"){
   LocalDateTime nextWindow = getNextMaintenanceWindowStart()
   String nextWindowDateString = nextWindow.format(DateTimeFormatter.ISO_LOCAL_DATE)
   List issues = jira_query("project=\"${project}\" AND summary ~ '${nextWindowDateString} Maintenance Window'")
-  return issues.size > 0
+  Boolean issueExists = issues.size > 0
+  if (issueExists){
+    print("Issue Exists for next maintenance window: ${issues[1]}")
+  }
+  return issueExists
+}
+
+/**
+* Jobs matching the patterns in this function are allowed to run during
+* a maintenance window to facilitate testing. Note this only overrides
+* the abort check in standard jobs, it won't override quietDown mode.
+*/
+@NonCPS
+Boolean allowBuildDuringMaintenanceWindow(){
+  patterns = [
+    /RE-Maintenance/,
+    /RE-unit-test/,
+    /RPC-Gating-Unit-Tests/,
+    /scatchpipeline/
+  ]
+
+  for (pattern in patterns){
+    if (env.JOB_NAME.contains(pattern)){
+      return true
+    }
+  }
+  print("This Job is not allowed to build during maintenance windows")
```

Comment:
You've added print statements to a number of these `Boolean` type functions for one value, why not add logging for the inverse result? Do the logs make it obvious so it's not required?

Reply:
The prints all show things that contribute to a build being aborted. A test that allows the build to continue doesn't need to log IMO.